### PR TITLE
Git initial configuration

### PR DESCRIPTION
**Pull request by HarryPerez**
## Summary
Git initial configuration. 

#### Features:
* New branch `repository-initiation`
* Setting remote url to `origin`
* Bootstrap `Initial commit`
* Branching from branch `master` of the repository.
* `validation` of repository url

### CODE REVIEW

File: generators/app/tasks/gitConfig.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        successMessage: 'Git pull of master successfully',
+        failureMessage: 'Git pull of master has failed'
+      })
+    )
+    .then(() =>
+      runCommand({
+        command: ['rm', ['README.md'], { cwd: `${process.cwd()}/${this.projectName}` }],
+        loadingMessage: 'Removing default README.md',
+        successMessage: 'Default README.md removed successfully',
+        failureMessage: 'Removing of default README.md has failed'
```
Comment (pabloccid) [inline generators/app/tasks/gitConfig.js]: Be aware that if readme.md cant be deleted by permissions, then the rest of the commands wont be executed. Maybe you could split the readme actions and the git actions in two different "promise chain" (i mean two different consecutive "then"s chains). 
Finishing the git configuration is far more important than using the repo url in the readme, and doing this last action may bring some permission trouble, based on the folder configuration.

File: generators/app/prompts.js [hunk 2]
```diff
@@ -14,10 +14,21 @@ module.exports = [
...
+    message: 'Do you want to configure a Github repo?',
+    required: true
+  },
+  {
+    when: values => values.configureGit,
     type: 'input',
     name: 'repoUrl',
-    message: 'What is the git repo url for this project?',
-    required: true
+    message: 'Enter your repo url in order to clone with SSH',
```
Comment (damfinkel) [inline generators/app/prompts.js]: I'd change to `Enter you repo SSH url in order to clone with SSH` so it's clear which url to pass

File: generators/app/index.js [hunk 3]
```diff
@@ -57,7 +58,7 @@ class GeneratorReact extends Generator {
 
   end() {
     if (this.configureGit) {
-      gitConfig.bind(this)();
+      configGit.bind(this);
```
Comment (pabloccid) [inline generators/app/index.js]: What if `gitInitiation` failed? Should this be ran anyway?

File: generators/app/prompts.js [hunk 4]
```diff
@@ -14,10 +14,21 @@ module.exports = [
...
+    message: 'Do you want to configure a Github repo?',
+    required: true
+  },
+  {
+    when: values => values.configureGit,
     type: 'input',
     name: 'repoUrl',
-    message: 'What is the git repo url for this project?',
-    required: true
+    message: 'Enter you repo SSH url in order to clone with SSH',
```
Comment (pabloccid) [inline generators/app/prompts.js]:
```suggestion
    message: 'Enter your repo SSH url in order to clone with SSH',
```

File: generators/app/index.js [hunk 5]
```diff
@@ -33,17 +34,20 @@ class GeneratorReact extends Generator {
...
-      configPackageJson.bind(this)();
-      copyTemplateFiles.bind(this)();
-    });
+    return Promise.resolve()
+      .then(() => {
+        this.log('Copying base project files...');
+        configPackageJson.bind(this)();
+        copyTemplateFiles.bind(this)();
+      })
+      .then(this.configureGit && configGit.bind(this));
```
Comment (pabloccid) [inline generators/app/index.js]: What if `gitInitiation` failed? Should this be ran anyway?

**Review by pabloccid — APPROVED** [review]

**Review by pabloccid — APPROVED** [review]

**Pull request merged**